src/main/java/io/apimatic/core/ErrorCase.java

```diff
@@ -41,7 +62,7 @@ private ErrorCase(final String reason, final ExceptionCreator<ExceptionType> exc
      * @throws ExceptionType Represents error response from the server.
      */
     public void throwException(Context httpContext) throws ExceptionType {
-        throw exceptionCreator.apply(reason, httpContext);
+        throw exceptionCreator.apply(getReason(httpContext), httpContext);
```

Comment:
this function should also hold the logic of getReason and replacePlaceHolder

Reply:
Done